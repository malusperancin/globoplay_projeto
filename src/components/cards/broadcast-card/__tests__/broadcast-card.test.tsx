/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/no-node-access */
import React from "react";
import { act, fireEvent, render } from "@testing-library/react";
import BroadcastCardType from "../../../cards/broadcast-card/broadcast-card-type";
import BroadcastCard from "../broadcast-card";

describe("BroadcastCard", () => {
  const simulateKeyRight = (element = window) => {
    fireEvent.keyDown(element, 40); // right 39  // down 40
  };

  const cardLive: BroadcastCardType = {
    id: "teste",
    subtitle: "teste",
    headline: "teste",
    title: "teste",
    posterUrl: "teste",
    backgroundUrl: "teste",
    selfRatedContent: true,
    contentRating: "teste",
    stringTime: "teste",
    progress: 0,
    broadcast: {
      isLive: true,
      channel: {
        __typename: "BroadcastChannel",
        name: "teste",
        logo: "teste",
        logoTrimmed: "teste",
        logoSelected: "teste",
      },
      mediaId: "teste",
      logo: "teste",
      logoSelected: "teste",
      media: {
        availableFor: "teste",
        serviceId: "teste",
      },
    },
    UI_number: 1,
    categories: [],
  };

  const cardNotLive: BroadcastCardType = {
    id: "teste",
    subtitle: "teste",
    headline: "teste",
    title: "teste",
    posterUrl: "teste",
    backgroundUrl: "teste",
    selfRatedContent: true,
    contentRating: "teste",
    stringTime: "teste",
    progress: 0,
    broadcast: {
      isLive: false,
      channel: {
        __typename: "BroadcastChannel",
        name: "teste",
        logo: "teste",
        logoTrimmed: "teste",
        logoSelected: "teste",
      },
      mediaId: "teste",
      logo: "teste",
      logoSelected: "teste",
      media: {
        availableFor: "teste",
        serviceId: "teste",
      },
    },
    UI_number: 0,
    categories: [],
  };

  const broadcastLiveComponent = (
    <BroadcastCard data={cardLive} isFocus={true} translate={100} index={0} />
  );

  const broadcastNotLiveComponent = (
    <BroadcastCard
      data={cardNotLive}
      isFocus={true}
      translate={100}
      index={0}
    />
  );

  it("should render without throwing an error", () => {
    const { container } = render(broadcastLiveComponent);
    expect(container).toBeTruthy();
  });

  it("should display live icon", () => {
    const { getByTestId } = render(broadcastLiveComponent);

    const liveIcon =
      getByTestId("broadcast-card-0").getElementsByClassName("liveIcon")[0];

    expect(liveIcon).toBeTruthy();
  });

  it("should not display live icon", () => {
    const { getByTestId } = render(broadcastNotLiveComponent);

    const liveIcon =
      getByTestId("broadcast-card-0").getElementsByClassName("liveIcon")[0];

    expect(liveIcon).toBeFalsy();
  });
});
