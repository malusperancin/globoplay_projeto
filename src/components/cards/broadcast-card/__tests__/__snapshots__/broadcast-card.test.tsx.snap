// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BroadcastCard should render without throwing an error 1`] = `
<div>
  <div
    class="container"
    data-testid="broadcast-card"
  >
    <button
      class="buttonFocus"
      style="transform: translate(100px);"
      type="button"
    >
      <div
        class="logoContainer"
      >
        <img
          alt=""
          class="logoChannel"
          src="teste"
        />
      </div>
      <img
        alt=""
        class="poster"
        src="teste"
      />
      <div
        class="info"
      >
        <div
          class="firstLine"
        >
          <img
            alt="ao vivo"
            class="liveIcon"
            src="live-icon.png"
          />
          <p
            class="hour"
          >
             
            teste
             
          </p>
        </div>
        <p
          class="titleSelect"
        >
          teste
        </p>
      </div>
    </button>
  </div>
</div>
`;
